apps/sms/test/marionette/conversation_test.js

```diff
@@ -556,4 +567,106 @@ marionette('Conversation Panel Tests', function() {
+      assert.lengthOf(recipients, 2, 'There should be 2 recipients');
+      assert.include(recipients, '+123', '+123 should be in the recipients');
+      assert.include(recipients, '+234', '+234 should be in the recipients');
+      assert.equal(newMessageView.messageText, '', 'MMS text should be empty');
+    });
+
+    test('Sms being written: message and recipients should be kept',
+    function() {
+      conversationView.typeMessage('message2');
+      var newMessageView = conversationView.includeSomeoneElse();
+      assert.equal(newMessageView.recipients, '+123',
+        'Recipient should be +123');
+      assert.equal(newMessageView.messageText, 'message2',
+        'message text should be kept');
+    });
+
+    test('Mms being written: all info must be kept', function() {
+      conversationView.typeMessage('message2');
+      // add a subject
+      conversationView.showSubject();
+      conversationView.typeSubject('subject1');
+      var newMessageView = conversationView.includeSomeoneElse();
+
+      // assert that we kept all the infos
+      assert.equal(newMessageView.recipients, '+123',
+        'Recipient should be +123');
+      assert.equal(newMessageView.messageText, 'message2',
+        'Message text should be kept');
+      assert.equal(newMessageView.subject, 'subject1',
+        'Subject should be kept');
+
+      // let's add a recipients and send the message.
+      // we will have a thread with several recipients
+      newMessageView.addNewRecipient('+234');
+      conversationView = newMessageView.send();
+      // write another MMS
+      conversationView.typeMessage('message3');
+      conversationView.showSubject();
+      conversationView.typeSubject('subject2');
+      // add a blob
+      conversationView.addAttachment().choose('Messages Activity Caller')
+        .pickImage();
+      messagesApp.switchTo();
+      // take screenshot of current attachment to compare it later
+      var beforeScreenshot = conversationView
+        .takeComposerAttachmentScreenshot(0);
+
+      // redirect to new message to add recipients
+      newMessageView = conversationView.includeSomeoneElse();
+
+      // assert the content
+      var recipients = newMessageView.recipients;
+      assert.lengthOf(recipients, 2, 'There should be 2 recipients');
+      assert.include(recipients, '+123', '+123 should be in the recipients');
+      assert.include(recipients, '+234', '+234 should be in the recipients');
+      assert.equal(newMessageView.messageText, 'message3',
+        'Message text should be kept');
+      assert.equal(newMessageView.subject, 'subject2',
+        'Subject should be kept');
+      // Tap the header to hide the keyboard: when it is displayed, the
+      // attachment is hidden, which prevents the screenshot to be taken
+      newMessageView.tapHeader();
+      client.waitFor(function() {
+        return beforeScreenshot ==
+          newMessageView.takeComposerAttachmentScreenshot(0);
```

Comment:
I have something for you :D
https://github.com/mozilla-b2g/gaia/blob/002fc618166f2887753b254a8b5e7d6195a86ed8/apps/keyboard/test/marionette/dismiss_test.js#L52-L61

Basically:
* I'd move `longPressSpaceBar` in lib/keyboard so that you could import it here (this is a little more explicit than tapHeader); but this is optional, as this is quite more work.
* it uses something from the system lib (note: `client.loader.getAppClass` is like a clever `require('../../../etc')`, defined in https://github.com/mozilla-b2g/gaia/blob/002fc618166f2887753b254a8b5e7d6195a86ed8/shared/test/integration/marionette_loader.js#L36-L43). Note that the test code here is quite obsolete, as now we have https://github.com/mozilla-b2g/gaia/blob/002fc618166f2887753b254a8b5e7d6195a86ed8/apps/system/test/marionette/lib/input_management.js#L62. Maybe you could add a `waitForKeyboardFrameHidden` just like we have `waitForKeyboardFrameDisplayed` already, and use it directly in this test.